source/ada/lsp-ada_contexts.adb

```diff
@@ -339,18 +368,69 @@ package body LSP.Ada_Contexts is
    begin
       Self.Documents.Delete (Item.uri);
       Free (Document);
+
+      declare
+         Name : constant LSP.Types.LSP_String := Self.URI_To_File (Item.uri);
+         File : constant Virtual_File := Create
+           (Filesystem_String (LSP.Types.To_UTF_8_String (Name)),
+            Normalize => True);
+      begin
+         if Self.Extra_Files.Contains (File) then
```

Comment:
Can you please a comment of what you are doing here? Generallu speaking I think that you should comment a bit more your code, not only the subprogram declarations

Reply:
done